OKR.md

```diff
@@ -2,6 +2,30 @@
 
 We try to frame our ongoing work using a process based on quarterly Objectives and Key Results (OKRs). Objectives reflect outcomes that are challenging, but realistic. Results are tangible and measurable.
 
+## 2018 Q3
```

Comment:
Let's continue discussing the OKRs over the markdown, taking inspiration from the notes already shared in this thread.

I've defined 3 main objectives (rewording is welcome if you see the need) for this quarter which capture the most requested and needed features and bug fixes.

Let's shoot to get the final list of KRs by Wednesday, final review on Thursday and ship it on Friday.

Reply:
@alanshaw @hugomrdias @achingbrain. I went again through your list of suggestions and completed the list of KRs. Please review:

- Check if wording makes sense.
- Check if they are achievable this quarter.
- Check the ones you will want to take the lead.

I've assigned some KRs already, given that they are a continuation of previous work.

Reply:
Do you think it's worth having a 'technical debt' section for things similar to the CI testing time item?

Feature parity with the go API would also be a good thing to shoot for.

Reply:
> What features (APIs) do you feel js-ipfs urgently needs

The biggest pain for me at the moment is the [DAG API being incomplete](https://github.com/ipfs/js-ipfs/blob/master/test/http-api/interface.js#L21).

Also `npm-on-ipfs` needs [`ipfs.name.resolve`](https://github.com/ipfs-shipyard/npm-on-ipfs/blob/master/src/registry-cache.js#L19) and [`ipfs.name.publish`](https://github.com/ipfs-shipyard/npm-on-ipfs/blob/master/src/registry-cache.js#L56) which I don't think are implemented yet although I guess that would be captured by the 'make npm-on-ipfs a thing' OKR.

Reply:
@achingbrain IPNS is landing this quarter by the libp2p team. Note that publishing and resolving names is for sharing the registry only, all the rest should work without it which is the bulk of the challenge.